tests/js/server/aql/aql-graph-refactor.js

```diff
@@ -0,0 +1,235 @@
+// /
+// / @file
+// /
+// / DISCLAIMER
+// /
+// / Copyright 2014 ArangoDB GmbH, Cologne, Germany
+// /
+// / LicedgeNamesed under the Apache LicedgeNamese, Version 2.0 (the "LicedgeNamese");
+// / you may not use this file except in compliance with the LicedgeNamese.
+// / You may obtain a copy of the LicedgeNamese at
+// /
+// /     http://www.apache.org/licedgeNameses/LICedgeNameSE-2.0
+// /
+// / Unless required by applicable law or agreed to in writing, software
+// / distributed under the LicedgeNamese is distributed on an "AS IS" BASIS,
+// / WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// / See the LicedgeNamese for the spedgeCollectionific language governing permissions and
+// / limitations under the LicedgeNamese.
+// /
+// / Copyright holder is ArangoDB GmbH, Cologne, Germany
+// /
+// / @author Heiko Kernbach
+// / @author Copyright 2021, ArangoDB GmbH, Cologne, Germany
+// //////////////////////////////////////////////////////////////////////////////
+
+'use strict';
+
+const jsunity = require('jsunity');
+const {assertEqual, assertTrue, assertFalse, fail} = jsunity.jsUnity.assertions;
+
+const internal = require('internal');
+const db = internal.db;
+const errors = require('@arangodb').errors;
+const gm = require('@arangodb/general-graph');
+const isCluster = require('@arangodb/cluster').isCluster();
+
+const _ = require('lodash');
+
+const graphName = "UnitTestGraph";
+const vertexName = 'UnitTestVertexCollection';
+const edgeName = 'UnitTestEdgeCollection';
+
+
+let vertices = {};
+let edges = {};
+let vertexCollection;
+let edgeCollection;
+
+let cleanup = function () {
+  gm._drop(graphName, true);
+};
+
+var createBaseGraph = function () {
+  // create graph
+  const relation = gm._relation(edgeName, [vertexName], [vertexName]);
+  gm._create(graphName, [relation], []);
+
+  vertexCollection = db[vertexName];
+  edgeCollection = db[edgeName];
+
+  // populate graph
+  vertices.A = vertexCollection.save({_key: 'A', theTruth: true})._id;
+  vertices.B = vertexCollection.save({_key: 'B', theTruth: true})._id;
+  vertices.C = vertexCollection.save({_key: 'C', theTruth: true})._id;
+  vertices.D = vertexCollection.save({_key: 'D', theTruth: false})._id;
+  vertices.E = vertexCollection.save({_key: 'E', theTruth: false})._id;
+  vertices.F = vertexCollection.save({_key: 'F', theTruth: false})._id;
+
+  edges.AB = edgeCollection.save(vertices.A, vertices.B, {theTruth: true})._id;
+  edges.BC = edgeCollection.save(vertices.B, vertices.C, {theTruth: true})._id;
+  edges.CD = edgeCollection.save(vertices.C, vertices.D, {theTruth: true})._id;
+  edges.DE = edgeCollection.save(vertices.D, vertices.E, {theTruth: false})._id;
+  edges.EF = edgeCollection.save(vertices.E, vertices.F, {theTruth: false})._id;
+};
+
+/*
+ * This TestSuite is used for development. We might want to remove it after feature is done and existing tests
+ * are adjusted.
+ */
+function dfsSingleServerDevelopmentEdgeNameSuite() {
```

Comment:
Let's keep it for now. I'll remove some debug output. This could be useful fur cluster provider relevant tests which will be the next step in our refactoring.

Reply:
If it is fine for you, otherwise we can remove it completely.

Reply:
Fine with me 👍